src/test/java/views/file_session_storage/FileSessionStorageTest.java

```diff
@@ -0,0 +1,59 @@
+package views.file_session_storage;
+
+
+import entities.SessionStorage;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+
+/**
+ * Tests the FileSessionStorage class
+ */
+class FileSessionStorageTest {
+    static SessionStorage sampleSession;
+
+    @BeforeAll
+    public static void SessionLoadUCICreateSampleSession() {
+        sampleSession = new SessionStorage();
+        sampleSession.addRecurExpense(new entities.Expense("abc", new entities.Category("lala", 10.0), 3.0));
+        sampleSession.addMonth(new entities.MonthlyStorage(1, 20.0));
+    }
+
+
+    /**
+     * Tests the saving and loading functionality of FileSessionStorage
+     * This isn't split into two test cases because they are heavily intertwined; we can't test loading
+     * a file without saving one first too.
+     */
+    @Test
+    public void FileSessionStorage() {
+        FileSessionStorage fileSessionStorage = new FileSessionStorage();
+
+        // Save session into file
+        try {
+            fileSessionStorage.save("FSSSaveFileTest.ser", sampleSession);
+        } catch (Exception e) {
+            Assertions.fail();
```

Comment:
It is a little unclear what Assertions.fail() is testing for. Maybe try to specify an exception? or explain in more detail in Javadoc. (this also includes the ones below)

Reply:
Yes, I will rewrite the testing for FileSessionStorage to make it clearer, and also I needed to add tests for the fail cases

Reply:
Done